### PR TITLE
Fix #100: Use batchUpdate in transaction

### LINKED ISSUE #100: Use batchUpdate in transaction
Methods `DatabaseAuditWriter::flush` and `DatabaseAuditWriter::cleanup` should be transactional.

### PR DESCRIPTION
**Pull request by romanstrobl**

### CODE REVIEW

File: audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java [hunk 1]
```diff
@@ -24,21 +26,27 @@ public class ClassUtil extends SecurityManager {
...
+     * @param packageFilter Packages to filter out when resolving calling class.
      * @return Calling class.
      */
-    public static Class<?> getCallingClass(String packageFilter) {
+    public static Class<?> getCallingClass(List<String> packageFilter) {
         final Class<?>[] trace = INSTANCE.getClassContext();
         if (trace == null) {
             return null;
         }
+        classLoop:
```
Comment (petrdvorak) [inline audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java:37]: Not a big fan of this... could we do it without the label?

**Review by petrdvorak — COMMENTED** [review]
Not sure about using label - it feels a bit archaic...

File: audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java (hunk 1 above)
Comment (petrdvorak) [reply audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java:37]: Btw this is one of the few examples where "Stream API" would work nicely for me.

Comment (romanstrobl) [reply audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java:37]: Sure, why not, I will change it.

**Review by petrdvorak — APPROVED** [review]
Looks good

File: audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java [hunk 2]
```diff
@@ -24,23 +30,27 @@ public class ClassUtil extends SecurityManager {
 
     /**
      * Get calling class.
-     * @param packageFilter Package to filter out when resolving calling class.
+     * @param packageFilter Packages to filter out when resolving calling class.
      * @return Calling class.
      */
-    public static Class<?> getCallingClass(String packageFilter) {
+    public static Class<?> getCallingClass(List<String> packageFilter) {
```
Comment (banterCZ) [inline audit-base/src/main/java/com/wultra/core/audit/base/util/ClassUtil.java:34]: You may just make this final

**Review by banterCZ — APPROVED** [review]

**Pull request merged**